### PR TITLE
DPL-1316 Refactor presigned url lambda to use shared path library

### PR DESCRIPTION
**Pull request by MatMoore**
This refactors the presigned URL lambda to use the new path construction module in our python base image. (see https://github.com/ministryofjustice/data-platform/pull/1384)

The goal of this refactor is that all the path construction logic is in one place so it's easier to change.

### CODE REVIEW

File: containers/daap-presigned-url/tests/unit/mock_test.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    monkeypatch.setattr(boto3, "client", lambda _name: s3_client)
+    monkeypatch.setenv("BUCKET_NAME", bucket_name)
+    monkeypatch.setattr(uuid, "uuid4", lambda: a_uuid)
+
+    # Emulate the data product being set up in the bucket
+    s3_client.create_bucket(
+        Bucket=bucket_name,
+        CreateBucketConfiguration={"LocationConstraint": region_name},
+    )
+    s3_client.put_object(Bucket=bucket_name, Key=f"code/{database}/hello", Body="")
```
Comment (PriyaBasker23) [inline containers/daap-presigned-url/tests/unit/mock_test.py]: Path doesnt seem relavant

**Review by PriyaBasker23 — DISMISSED** [review]

**Review by jacobwoffenden — DISMISSED** [review]

**Comment by MatMoore** [comment]
Before merge: need to change the path being checked by the lambda to the metadata path instead of /code.

File: containers/daap-presigned-url/tests/unit/mock_test.py (hunk 1 above)
Comment (MatMoore) [reply containers/daap-presigned-url/tests/unit/mock_test.py]: @PriyaBasker23 I've now changed the check to the metadata path (v1 metadata file). Can you check the latest commit and reapprove if ok?

**Review by PriyaBasker23 — DISMISSED** [review]

File: containers/daap-presigned-url/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM ghcr.io/ministryofjustice/data-platform-daap-python-base:0.3.0
+FROM ghcr.io/ministryofjustice/data-platform-daap-python-base:0.5.0
```
Comment (jacobwoffenden) [inline containers/daap-presigned-url/Dockerfile]: Could you bump this to `1.0.0` please

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]

File: containers/daap-presigned-url/Dockerfile (hunk 2 above)
Comment (murdo-moj) [reply containers/daap-presigned-url/Dockerfile]: were we gonna pin this at `:latest` for some images while we're developing?

**Review by jacobwoffenden — APPROVED** [review]

**Review by PriyaBasker23 — APPROVED** [review]

**Pull request merged**